styled/styled.go

```diff
@@ -0,0 +1,101 @@
+package styled
+
+import (
+	"fmt"
+)
+
+type Style struct {
+	Foreground string
+	Background string
+	Bold       bool
+	Dim        bool
+	Italic     bool
+	Underlined bool
+	Blink      bool
+	Inverse    bool
+}
+
```

Comment:
This can be a variable. "`DefaultStyle`" is a better name; "empty" is normally understood to be `StructName{}` in Go.

Speaking of which, it probably makes sense to use the empty string for default colors; this makes `Style{}` a valid value, which is considered good practice in Go. But I am not requiring you to do this in this PR.

Reply:
I also thought about defining empty strings to be the default color but wasn't sure how to handle the style transformer names. They should still be called `default` and `bg-default ` but I had the thought that it could be inconsistent...

Should the representation of segments contain the default color?